erts/emulator/beam/erl_proc_sig_queue.c

```diff
@@ -1280,6 +1331,45 @@ erts_proc_sig_fetch__(Process *proc,
     }
 }
 
+static void
+sched_sig_notify(void *vssnp)
+{
+    ErtsSchedSignalNotify *ssnp = (ErtsSchedSignalNotify *) vssnp;
+    Process *proc = erts_proc_lookup(ssnp->pid);
+    if (proc) {
+        erts_aint32_t state = erts_atomic32_read_acqb(&proc->state);
+        int nmsig = ssnp->nmsig;
+        int msig = ssnp->msig;
+        ASSERT(nmsig || msig);
+        if ((!!nmsig) & ((!(state & (ERTS_PSFLG_SIG_Q
+                                     | ERTS_PSFLG_NMSG_SIG_IN_Q)))
+                         | (!!(state & ERTS_PSFLG_ACTIVE_SYS)))) {
+            /*
+             * Either already handled or someone else set the active-sys
+             * flag...
+             */
+            nmsig = 0;
+        }
+        if ((!!msig) & (!(state & ERTS_PSFLG_ACTIVE))) {
+            /*
+             * Someone else set the active flag (we cannot determine if it
+             * has been handled or not by looking at the state flag)...
```

Comment:
Should it be
```
if ((!!msig) & (!!(state & ERTS_PSFLG_ACTIVE))) {
                ^^
```
that is, the active flag is already set.

Reply:
Yes, thanks! Fixed!